lib/SimpleHttpClient/SslClientConnection.cpp

```diff
@@ -343,15 +377,44 @@ bool SslClientConnection::connectSocket() {
 
   ERR_clear_error();
 
-  int ret = SSL_connect(_ssl);
+  int ret = -1;
+  int errorDetail = -1;
+
+  if (_isSocketNonBlocking) {
+    auto start = std::chrono::steady_clock::now();
+    while ((ret = SSL_connect(_ssl)) == -1) {
```

Comment:
NIT according to https://www.openssl.org/docs/man1.1.1/man3/SSL_connect.html the values should be "<0" not necessary "-1"